src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -205,14 +205,14 @@ void BaseCouplingScheme::receiveData(const m2n::PtrM2N &m2n, const DataMap &rece
 void BaseCouplingScheme::receiveDataForWindowEnd(const m2n::PtrM2N &m2n, const DataMap &receiveData)
 {
   // buffer current time
-  const double oldTime = getTime();
+  const double oldProgress = getTimeWindowProgress();
   // set _time state to point to end of this window such that _time in receiveData is at end of window
-  _time(_nextTimeWindowSize);
+  _timeWindowProgress(_nextTimeWindowSize);
   // receive data for end of window
   this->receiveData(m2n, receiveData);
   // reset time state;
-  _time = KahanAccumulator{};
-  _time(oldTime);
+  _timeWindowProgress = KahanAccumulator{};
+  _timeWindowProgress(oldProgress); // This should be 0, right?
```

Comment:
Yes. Otherwise, I think it would also be better to not initialize the Kahan accumulator with `double oldProgress`, but instead recover the actual `KahanAccumulator _timeWindowProgress` from above.

Maybe add an assertion to be sure before overwriting `_timeWindowProgress`: ``PRECICE_ASSERT(math::equals(getTimeWindowProgress(), 0.0));`

Reply:
This assertion is triggered straight away.